src/gc/impl/conservative/gc.d

```diff
@@ -2942,20 +2949,40 @@ struct Gcx
 
     version (Posix)
     {
-        // make sure the threads and event handles are reinitialized in a fork
-        __gshared Gcx* forkedGcx;
+        // A fork might happen while GC code is running in a different thread.
+        // Because that would leave the GC in an inconsistent state,
+        // make sure no GC code is running by acquiring the lock here,
+        // before a fork.
+
+        extern(C) static void _d_gcx_atfork_prepare()
+        {
+            if (instance)
+                ConservativeGC.lockNR();
+        }
+
+        extern(C) static void _d_gcx_atfork_parent()
+        {
+            if (instance)
+                ConservativeGC.gcLock.unlock();
+        }
```

Comment:
I will probably move these out to a general Druntime fork handler which also takes care of the thread stuff.

Reply:
That probably won't work as the GC can be swapped out at link time.

Reply:
Not sure what you mean. The atfork notifications would then be part of the abstract GC interface.